core/src/classes/codeConfig.ts

```diff
@@ -34,7 +34,9 @@ export interface ConfigurationObject {
   identifying?: boolean;
```

Comment:
This `ConfigurationObject` is getting out of hand!  This was a lazy choice on my part :( 
It might be better to make the type from the union of a few distinct types.  This way, we can check that each form of using the object is internally consistent. 

```ts
export interface ProfileConfigurationObject = {
// just what profile needs
}

export interface AppConfigurationObject = {
// just what app needs
}

export interface ConfigurationObject = ProfileConfigurationObject & AppConfigurationObject


```

Reply:
This could be another story

Reply:
Added a story to maybe delete the `SyncTable` part of this (https://www.pivotaltracker.com/story/show/179063948)

Reply:
I definitely agree! At the moment it's tough to know which part of that corresponds to each config object type. I think we should get this in and add another story for this, maybe also using these new interfaces in the config loaders as well. I can work on this right after if prioritized.

Reply:
I've got a PR up for this - https://github.com/grouparoo/grouparoo/pull/2118, but I won't merge it until after this!